server/controllers/authController.js

```diff
@@ -133,4 +135,100 @@ exports.authorize = (...roles) => {
         }
         next();
     }
+}
+
+//Send reset token to the email address of users who have forgotten their password
```

Comment:
Your `forgot password` and `reset password` are too long. They should only do one thing.

You are generating Tokens in those functions which should not be there. Have you heard about the `Single Responsibility Principle`

Your functions should do only one thing, so move those token generator functions outside and make it a helper function and any other function that should not be there